dotnet/src/IntegrationTests/Connectors/Weaviate/WeaviateMemoryStoreTests.cs

```diff
@@ -0,0 +1,282 @@
+﻿using System;
+using System.Linq;
+using System.Net.Http;
+using System.Threading.Tasks;
+using Microsoft.SemanticKernel.AI.Embeddings;
+using Microsoft.SemanticKernel.Connectors.Memory.Weaviate;
+using Microsoft.SemanticKernel.Connectors.Memory.Weaviate.Diagnostics;
+using Microsoft.SemanticKernel.Memory;
+using Xunit;
+
+namespace SemanticKernel.IntegrationTests.Connectors.Weaviate;
+
+/// <summary>
+///     Tests for <see cref="s_weaviateMemoryStore" /> collection and upsert operations.
+/// </summary>
+[Collection("Sequential")]
+public class WeaviateMemoryStoreTests
+{
+    private static readonly HttpClient s_httpClient = new()
+    {
+        DefaultRequestHeaders =
+        {
+            { "authorization", "my-secret-key" }
```

Comment:
nit: would be nice to add this to the configuration like with other integration tests. Another day, perhaps.

Reply:
I've removed this requirement and added a docker-compose file to help instantiate a local Weaviate instance.